teachopencadd/talktorials/T013_query_pubchem/talktorial.ipynb

```diff
@@ -9,7 +9,8 @@
     "Authors:\n",
```

Comment:
<p>This function is now more important than in the previous version of this code, so we should dedicate some more time to explain what's going on.</p><p> </p><p>Maybe we should split this function in four parts:</p><p> </p><ol><li>Create task / job so we obtain a job key -- say that this API is async</li><li>Check if the job finished -- details: we need to be nice and wait, explain why</li><li>Download results when finished (PubChem IDs)</li><li>Get the canonical Smiles from the PubChem IDs</li></ol><p> </p><p>Then we create the RDKit molecules and depict them.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/teachopencadd/pull/81/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Reply:
<p>I agree with @jaimergp here. Split your steps in stand-alone functions and all them step-by-step in this main function here.</p><p> </p><p>I'd probably move code to functions like this:</p><p><code>similar_compounds_from_smiles(smiles, threshold, n_records)</code></p><p><code>job_is_finished(job_id)</code></p><p><code>smiles_from_pubmed_cids(cids)</code></p><p> </p>

Reply:
<p>I split this function with three parts: </p><p>query_pubchem_for_similar_compounds(<em>smiles</em>,&nbsp;<em>threshold</em>=75,&nbsp;<em>n_records</em>=10)</p><p>check_and_download(<em>key</em>,&nbsp;<em>attempts</em>=25)</p><p>smiles_from_pubchem_cids(<em>cids</em>)</p><p> </p>

Reply:
<p>That sounds very good to me!</p><p> </p><p>Did you implement these functions already?</p><ul><li><code>check_and_download(<em>key</em>,&nbsp;<em>attempts</em>=25)</code></li><li><code>smiles_from_pubchem_cids(<em>cids</em>)</code></li></ul><p> </p><p>I can only find <code>query_pubchem_for_similar_compounds</code> . </p><p> </p><p>My apologies if I somehow missed that!</p>